src/west/manifest.py

```diff
@@ -1317,6 +1391,45 @@ def as_frozen_yaml(self, **kwargs) -> str:
     def projects(self) -> List[Project]:
         return self._projects
 
+    def is_active(self, project: Project,
+                  extra_groups: Optional[Iterable[str]] = None) -> bool:
+        '''Is a project active?
+
+        Projects with empty 'project.groups' lists are always active.
+
+        Otherwise, if any group in 'project.groups' is allowed by this
+        manifest's 'groups:' list (and the 'manifest.groups' local
+        configuration option, if we have a workspace), returns True.
+
+        Otherwise, i.e. if all of the project's groups are blocked,
+        this returns False.
+
+        "Inactive" projects should generally be considered absent from
+        the workspace for purposes like updating it, listing projects,
+        etc.
+
+        :param project: project to check
+        :param groups_extra: additional groups allow/block list, at
+            higher precedence than the manifest or configuration file
+        '''
+        if not project.groups:
+            # Projects without any groups are always active, so just
+            # exit early. Note that this happens to treat the
+            # ManifestProject as though it's always active. This is
+            # important for keeping it in the 'west list' output for
+            # now.
+            return True
+
+        if extra_groups:
+            blocked_groups = set(self._blocked_groups)
+            _update_blocked_groups(blocked_groups, extra_groups)
+        else:
+            blocked_groups = self._blocked_groups
+
+        # A project with groups is active if and only if any one of its
+        # groups are blocked.
```

Comment:
Did you mean, `groups are not blocked.` ?

Reply:
> Did you mean, `groups are not blocked.` ?

Yes. I just deleted the comment since the docstring has all the details anyway. Thanks.